tensorboard/plugins/text_v2/text_v2_plugin_test.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Integration tests for the Text Plugin."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import collections.abc
+import functools
+import os
+import textwrap
+import numpy as np
+import tensorflow as tf
+
+from tensorboard import plugin_util
+from tensorboard.backend.event_processing import data_provider
+from tensorboard.backend.event_processing import (
+    plugin_event_multiplexer as event_multiplexer,
+)
+from tensorboard.plugins import base_plugin
+from tensorboard.plugins.text_v2 import text_v2_plugin
+from tensorboard.util import test_util
+
+tf.compat.v1.disable_v2_behavior()
+
+GEMS = ["garnet", "amethyst", "pearl", "steven"]
+
+
+class TextPluginTest(tf.test.TestCase):
+    def setUp(self):
+        self.logdir = self.get_temp_dir()
+
+    def with_plugin(generate_testdata=True, include_text=True):
```

Comment:
The main point of this decorator was to support testing across both the bare multiplexer and the generic data provider; since we're not copying the support for the bare multiplexer, this decorator isn't adding any value here.

I'd just convert this into a regular test helper method that returns the initialized plugin:

```python
def create_plugin(generate_testdata=True, include_text=True):
    ...

# example usage
def testRoutesProvided(self):
    plugin = self.create_plugin()
    routes = plugin.get_plugin_apps()
    ...
```

You can at that point also remove the `with self.subTest()` call, since it was only useful when there were multiple calls to subTest but now there's just one.

Reply:
done